svc/WebSocketService.ts

```diff
@@ -39,6 +39,13 @@ export class WebSocketService extends HoistService {
     static instance: WebSocketService;
 
     readonly HEARTBEAT_TOPIC = 'xhHeartbeat';
+    /** Check connection and send a new heartbeat (which should be promptly ack'd) every 10s. */
+    readonly HEARTBEAT_INTERVAL_MS = 10 * SECONDS;
+    /**  If no heartbeat ack (or other msg) received for past 30s, force a reconnect attempt. */
+    readonly HEARTBEAT_ACK_TIMEOUT_MS = 30 * SECONDS;
+    /** But wait at least 30s after a heartbeat-driven reconnect attempt before trying again. */
+    readonly HEARTBEAT_MIN_RECONNECT_INTERVAL_MS = 30 * SECONDS;
```

Comment:
I added this out of an abundance of caution - we want to detect connectivity issues quickly with 10s heartbeat interval, and if there's an issue found we'll fire off a reconnect attempt.  But if 10s later we still have an issue, wait a bit longer before trying again - something is up...

Realizing this might be better as, say, 25s, if we wanted it to skip only two timer cycles then try again on the third.